extensions/arc/test-supplement/pom.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-arc-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-arc-test-supplement</artifactId>
```

Comment:
Hm, I wonder if this artifact will get released and deployed to the maven central as well?

Reply:
Probably. I was thinking I could abuse some other dependency of the ArC `deployment` module, like AssertJ, but that would require keeping the version string in the test in sync with the POM, which felt like a worse deal than an extra module with a single class.

Reply:
Maybe we could try to skip this module during a release. I've found https://stackoverflow.com/questions/31495316/skip-submodule-during-maven-release where the following config is used in the module that should be skipped:
```xml
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-deploy-plugin</artifactId>
  <configuration>
    <skip>true</skip>
  </configuration>
</plugin>

```

Reply:
That would work, but the ArC `deployment` module has a [test-scoped] dependency on this artifact, so `quarkus-arc-deployment` POM might cause errors. Is this tiny artifact a problem? I'd much rather keep it than to face surprises down the line.

It would be best if I could create an in-memory archive using ShrinkWrap and add it as a non-application archive, but `QuarkusUnitTest` doesn't offer an API for that and I don't really know if `QuarkusBootstrap.Builder` allows it or not. @aloubyansky would you please know if that would be feasible?

Reply:
> Is this tiny artifact a problem?

It is not. The central repo is probably full of other huge useless artifacts anyway :shrug:

Reply:
@Ladicek we are not going to add a dependency on ShrinkWrap to the bootstrap API, however there is `io.quarkus.maven.dependency.ResolvedDependencyBuilder` that allows you create a dependency with the paths you need.
There is also `io.quarkus.bootstrap.workspace.WorkspaceModule.builder()` API to initialize a project module with the sources and classes paths.
Would that help?

Reply:
Yes, you can also set it as `appArtifact`, depending on what you need.

Reply:
I tried to do something like

```java
ResolvedDependency dep = ResolvedDependencyBuilder.newInstance()
        .setGroupId(groupId)
        .setArtifactId(artifactId)
        .setVersion(version)
        .setResolvedPath(path)
        .setWorkspaceModule(WorkspaceModule.builder()
                .setModuleId(WorkspaceModuleId.of(groupId, artifactId, version))
                .setModuleDir(path)
                .setBuildDir(path)
                .build())
        .build();
```

(where `groupId`, `artifactId` and `version` are things I made up, and `path` is the directory where the `.class` file is), and when I add that `dep` to forced dependencies, Quarkus tries to download it from Central.

I also tried to add an `AdditionalDependency` with the `forceApplicationArchive` flag set to `false` to the set of additional dependencies, but it doesn't do what I need (probably is still considered an application archive).

Having an extra module in the ArC extension isn't super pretty, but works as I need, so I'll probably leave it be.

Reply:
It probably fails to resolve the POM artifact.

Reply:
It actually tries to download both the POM and the JAR:

```
Downloading from central: https://repo.maven.apache.org/maven2/com/example/test1540098276/1.0.0/test1540098276-1.0.0.pom
Downloading from central: https://repo.maven.apache.org/maven2/com/example/test1540098276/1.0.0/test1540098276-1.0.0.jar
```

:shrug: